set.mm

```diff
@@ -31488,12 +31488,29 @@ generally appear in a single form (either definitional, but more often
     $d A x y $.  $d ph x $.  $d ch y $.
     rabrabi.1 $e |- ( x = y -> ( ch <-> ph ) ) $.
     $( Abstract builder restricted to another restricted abstract builder with
-       implicit substitution.  (Contributed by AV, 2-Aug-2022.)  Avoid ~ ax-10
-       and ~ ax-11 .  (Revised by Gino Giotto, 20-Aug-2023.) $)
+       implicit substitution.  (Contributed by AV, 2-Aug-2022.)  Avoid
+       ~ ax-10 , ~ ax-11 and ~ ax-12 .  (Revised by Gino Giotto,
+       12-Oct-2024.) $)
     rabrabi $p |- { x e. { y e. A | ph } | ps } = { x e. A | ( ch /\ ps ) } $=
+      ( wa crab cv wcel cab df-rab eleq2i wsb df-clab weq eleq1w wb bitri anass
+      bicomd equcoms anbi12d sbievw anbi1i rabbia2 ) BCBHZDAEFIZFDJZUIKZBHUJFKZ
+      CHZBHULUHHUKUMBUKUJEJFKZAHZELZKZUMUIUPUJAEFMNUQUOEDOUMUODEPUOUMEDEDQUNULA
+      CEDFRACSDEDEQCAGUBUCUDUETTUFULCBUATUG $.
+    $( $j usage 'rabrabi' avoids 'ax-10' 'ax-11' 'ax-12'; $)
+  $}
+
+  ${
+    $d A x y $.  $d ph x $.  $d ch y $.
+    rabrabiOLD.1 $e |- ( x = y -> ( ch <-> ph ) ) $.
+    $( Obsolete version of ~ rabrabi as of 12-Oct-2024.  (Contributed by AV,
+       2-Aug-2022.)  Avoid ~ ax-10 and ~ ax-11 .  (Revised by Gino Giotto,
+       20-Aug-2023.)  (Proof modification is discouraged.)
+       (New usage is discouraged.) $)
+    rabrabiOLD $p |-
+                   { x e. { y e. A | ph } | ps } = { x e. A | ( ch /\ ps ) } $=
```

Comment:
Proof is much shorter. We should probably keep it as ALT.

Reply:
I have a proposal:

Observing that [rabrab](https://us.metamath.org/mpeuni/rabrab.html) uses [rabid](https://us.metamath.org/mpeuni/rabid.html) we could add `rabidw` and use it in [rabrabi](https://us.metamath.org/mpeuni/rabrabi.html):

<pre>
  ${
    $d x y $.  $d ph x $.  $d ps y $.  $d A y $.
    rabidw.1 $e |- ( x = y -> ( ph <-> ps ) ) $.
    $( Version of ~ rabid using implicit subsitution, which requires fewer
       axioms.  (Contributed by GG, 13-Oct-2024.) $)
    rabidw $p |- ( x e. { y e. A | ph } <-> ( x e. A /\ ps ) ) $=
      ( cv crab wcel wa cab wsb df-rab eleq2i df-clab eleq1w wb equcoms anbi12d
      weq sbievw 3bitri ) CGZADEHZIUCDGEIZAJZDKZIUFDCLUCEIZBJZUDUGUCADEMNUFCDOU
      FUIDCDCTUEUHABDCEPABQCDFRSUAUB $.
    $( $j usage 'rabidw' avoids 'ax-10' 'ax-11' 'ax-12'; $)
  $}
</pre>

<pre>
  ${
    $d x y $.  $d A y $.  $d ph x $.  $d ch y $.
    rabrabi.1 $e |- ( x = y -> ( ph <-> ch ) ) $.
    $( Abstract builder restricted to another restricted abstract builder with
       implicit substitution.  (Contributed by AV, 2-Aug-2022.)  Avoid
       ~ ax-10 , ~ ax-11 and ~ ax-12 .  (Revised by Gino Giotto,
       12-Oct-2024.) $)
    rabrabi $p |- { x e. { y e. A | ph } | ps } = { x e. A | ( ch /\ ps ) } $=
      ( wa crab cv wcel rabidw anbi1i anass bitri rabbia2 ) BCBHZDAEFIZFDJZRKZB
      HSFKZCHZBHUAQHTUBBACDEFGLMUACBNOP $.
    $( $j usage 'rabrabi' avoids 'ax-10' 'ax-11' 'ax-12'; $)
  $}
</pre>

This would also allow to remove the DV condition between x and A in [rabrabi](https://us.metamath.org/mpeuni/rabrabi.html) (notice that I removed it above). I would also change the hypothesis `rabrabi.1 $e |- ( x = y -> ( ch <-> ph ) ) $.` to `rabrabi.1 $e |- ( x = y -> ( ph <-> ch ) ) $.` to make [rabrabi](https://us.metamath.org/mpeuni/rabrabi.html)'s proof even shorter. This implies adding a [equcoms](https://us.metamath.org/mpeuni/equcoms.html) step in [wlksnwwlknvbij](https://us.metamath.org/mpeuni/wlksnwwlknvbij.html) (which is the only theorem using [rabrabi](https://us.metamath.org/mpeuni/rabrabi.html)), but in my opinion it's better if `ph` and `ch` are placed in the same order as in the statement, as it is standard practice across the database.

Reply:
I had a look and it's probably too much effort for that single theorem. Actually, the OLD can stay OLD as you first proposed.